merlion/models/forecast/autoregressivebase.py

```diff
@@ -0,0 +1,251 @@
+        """
+        1-step auto-regression method for multivariate data, each regression step updates one data point for each sequence
+        :return: pred of target_seq_index for steps [n_samples, steps]
+        """
+
+        if steps is None:
+            steps = self.max_forecast_steps
+
+        inputs = np.atleast_2d(inputs)
+
+        pred = np.empty((len(inputs), steps))
+
+        for i in range(steps):
+            # next forecast shape: [n_samples, self.dim]
+            next_forecast = self.model.predict(inputs)
+            pred[:, i] = next_forecast[:, self.target_seq_index]
+            if i == steps - 1:
+                break
+            inputs = self._update_prior_nd(inputs, next_forecast)
+        return pred
+
+    def _update_prior_nd(self, prior: np.ndarray, next_forecast: np.ndarray):
+        """
+        regressively update the prior by concatenate prior with next_forecast on the sequence dimension,
+        :param prior: the prior [n_samples, n_seq * maxlags]
+        :param next_forecast: the next forecasting result [n_samples, n_seq]
+        :return: updated prior
+        """
+        assert isinstance(prior, np.ndarray) and len(prior.shape) == 2
+        assert isinstance(next_forecast, np.ndarray) and len(next_forecast.shape) == 2
+
+        # unsqueeze the sequence dimension so prior and next_forecast can be concatenated along sequence dimension
+        # for example,
+        # prior = [[1,2,3,4,5,6,7,8,9], [10,20,30,40,50,60,70,80,90]], after the sequence dimension is expanded
+        # prior = [[[1,2,3], [4,5,6], [7,8,9]],
+        #          [[10,20,30],[40,50,60],[70,80,90]]
+        #         ]
+        # next_forcast = [[0.1,0.2,0.3],[0.4,0.5,0.6]], after the sequence dimension is expanded
+        # next_forecast = [[[0.1],[0.2],[0.3]],
+        #                  [[0.4],[0.5],[0.6]]
+        #                 ]
+        prior = prior.reshape(len(prior), self.dim, -1)
+        next_forecast = np.expand_dims(next_forecast, axis=2)
+        prior = np.concatenate([prior, next_forecast], axis=2)[:, :, -self.maxlags:]
+        return prior.reshape(len(prior), -1)
+
+    def _update_prior_1d(self, prior: np.ndarray, next_forecast: np.ndarray):
+        """
+        regressively update the prior by concatenate prior with next_forecast for the univariate,
+        :param prior: the prior [n_samples, maxlags]
+        :param next_forecast: the next forecasting result [n_samples, n_prediction_steps],
+            if n_prediciton_steps ==1, maybe [n_samples,]
+        :return: updated prior
+        """
+        assert isinstance(prior, np.ndarray) and len(prior.shape) == 2
+        assert isinstance(next_forecast, np.ndarray)
+
+        if len(next_forecast.shape) == 1:
+            next_forecast = np.expand_dims(next_forecast, axis=1)
+        prior = np.concatenate([prior, next_forecast], axis=1)[:, -self.maxlags:]
+        return prior.reshape(len(prior), -1)
```

Comment:
Can you consolidate these into a single function?

Reply:
Although I can combine them together, I feel it is good to keep both since they are having pretty different internal manipulation along different dimensions

Reply:
Done, add `for_univariate`